webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/Init.elm

```diff
@@ -0,0 +1,33 @@
+port module ComplianceMode.Init exposing (..)
+
+import Http exposing (Error)
+import Task
+import Json.Decode exposing (..)
+import Json.Decode.Pipeline as D exposing (..)
+
+import ComplianceMode.DataTypes exposing (..)
+
+
+-- PORTS / SUBSCRIPTIONS
+port saveMode            : Value  -> Cmd msg
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+  Sub.none
+
+init : { contextPath : String, hasWriteRights : Bool, complianceMode : {name : String, heartbeatPeriod : Int}, globalMode : {name : String, heartbeatPeriod : Int} } -> ( Model, Cmd Msg )
+init flags =
+  let
+    getComplianceMode : String -> ComplianceMode
+    getComplianceMode mode =
+      case mode of
+        "full-compliance"  -> FullCompliance
+        "changes-only"     -> ChangesOnly
+        "reports-disabled" -> ReportsDisabled
+        _                  -> UnknownMode
```

Comment:
This should either be an error, not ignored

Reply:
I used this "Error" mode to display an error in the form, but the user can still select a new valid report mode and save his changes.